cache/refs.go

```diff
@@ -268,3 +280,17 @@ func (sr *mutableRef) release(ctx context.Context) error {
 	// }
 	return nil
 }
+
+func setReadonly(mounts []mount.Mount) []mount.Mount {
```

Comment:
I guess this should be detected already in solver or llb parsing then as `readonly` type is a hint to the reference counting. If worker can't support it then client should not request it.